README.md

```diff
@@ -39,56 +39,69 @@ Features
-(Ctrl+1) to install the Tycho maven integration.
+Some editors or editor extensions bundle eclipse.jdt.ls or contain logic to install it. If that is the case, you only need to install the editor extension. For example for Visual Studio Code you can install the [Extension Pack for Java](https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-pack) and it will take care of the rest.
 
-4. At that point, some plug-ins should still be missing in order to build the project. You can either open `org.eclipse.jdt.ls.target/org.eclipse.jdt.ls.tp.target` in the Target Editor (which is the default editor) and click on `Set Target Platform`, or alternatively, open `Preferences > Plug-in Development > Target Platform` and select `Java Language Server Target Definition`). Eclipse will take some time to download all the required dependencies. It should then be able to compile all the projects in the workspace.
 
-Building from the command line
-----------------------------
-**Pre-requisite: Java 11 must be installed on your machine and accessible in the PATH.**
+Running from the command line
+------------------------------
+
+If you built eclipse.jdt.ls from source, `cd` into `./org.eclipse.jdt.ls.product/target/repository`. If you downloaded a milestone or snapshot build, extract the contents.
 
-The following command will install [Apache Maven](https://maven.apache.org/) if necessary, then build the server into the  `/org.eclipse.jdt.ls.product/target/repository` folder:
+To start the server in the active terminal, adjust the following command as described further below and run it:
 
 ```bash
-    $ ./mvnw clean verify
-````
+java \
+	-Declipse.application=org.eclipse.jdt.ls.core.id1 \
+	-Dosgi.bundles.defaultStartLevel=4 \
+	-Declipse.product=org.eclipse.jdt.ls.core.product \
+	-Dlog.level=ALL \
+	-noverify \
+	-Xmx1G \
+	-jar ./plugins/org.eclipse.equinox.launcher_1.5.200.v20180922-1751.jar \
+	-configuration ./config_linux \
+	-data /path/to/data \
+	--add-modules=ALL-SYSTEM \
+	--add-opens java.base/java.util=ALL-UNNAMED \
+	--add-opens java.base/java.lang=ALL-UNNAMED
+```
 
-Running from the command line
-------------------------------
-1. Choose a connection type from "Managing connection types" section below, and then set those environment variables in your terminal or specify them as system properties with `-D` prior to continuing
+1. Chose a value for `-configuration`: this is the path to your platform's configuration directory. For linux, use `./config_linux`. For windows, use `./config_win`. For mac/OS X, use `./config_mac`.
+2. Change the filename of the jar in `-jar ./plugins/...` to match the version you built or downloaded.
+3. Choose a value for `-data`: the value for your data directory, should be the directory where your active workspace is, and you wish for the java langserver to add in its default files. Should also be the absolute path to this directory, ie., /home/username/workspace
```

Comment:
From what I've seen this should be unique per project(?).
eclipse.jdt.ls always ends up creating a `jdt.ls-java-project` folder inside (instead of project specific ones).

Or is this due to the lsp client not setting some initialization options that it should?

Reply:
Yes, it's per workspace. `-data` parameter is used to specify a local cache directory to store the metadata files (such as index files) generated by Java language server. 

Please notice that this parameter is not the user's project root folder. The project root folder you want to import is passed by LSP Initialization options.

Reply:
Thanks for the clarification. I rephrased the third point here a bit in the fixup I pushed